test/CMakeLists.txt

```
@@ -1251,6 +1254,22 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_viirs_aod
                           -n 0.0"
                           viirs_aod.nc ${IODA_CONV_COMP_TOL})
 
+ecbuild_add_test( TARGET  test_${PROJECT_NAME}_viirs_smoke_dust
+                  TYPE    SCRIPT
+                  ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
+                  COMMAND bash
+                  ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                          netcdf
+                          "${Python3_EXECUTABLE} ${CMAKE_BINARY_DIR}/bin/viirs_aod2ioda.py
+                          -i testinput/JRR-AOD_v3r2_j01_s20240110T202044Z_e20240110T202044Z.sample.nc
+                          --adp_mask testinput/JRR-ADP_v3r2_j01_s20240110T202044Z_e20240110T202044Z.sample.nc
+                          --adp_conf_lvl 'medhigh'
+                          -o testrun/obs.20240110T202044Z_PT1M_aod.nc
+                          -m nesdis
+                          -k maskout
+                          -n 0.0"
```

I would advise modifying the code and simplify these, making a few more transparent `-n` should not have a shorthand and be --thinning_distance (option) and should have a default of 0.0.... the `-k` should be a logical switch, the `adp_conf_level` again spelled out a bit more `adp_confidence_level` and should default to `'medhigh'` and then does not need to be present in the call.  I can't think of a more verbose name for the `--adp_mask` but think it too should be more explicit.  And the help message should spell out the ADP acronym and have two section describing separately how to run for just AOD, and then how to run for ADP.

At this point OK doing 2 ctests one checking smoke and the other dust; however, this could be done via optional flag requiring two call to the converter

```suggestion
                          --adp_mask testinput/JRR-ADP_v3r2_j01_s20240110T202044Z_e20240110T202044Z.sample.nc
                          -o testrun/obs.20240110T202044Z_PT1M_aod.nc
                          --apply_mask"
```